src/ducks/payments/service.js

```diff
@@ -0,0 +1,75 @@
+import logger from 'cozy-logger'
+import { biApi } from './api'
+import { makePayload } from './helpers'
+import { waitForRealtimeEvent } from 'cozy-harvest-lib/dist/services/jobUtils'
+import { TRANSACTION_DOCTYPE } from 'doctypes'
+import get from 'lodash/get'
+
+const log = logger.namespace('payment-service')
+
+export const createBiPayment = async client => {
+  const jobResponse = await client.stackClient.jobs.create('konnector', {
+    mode: 'getPaymentToken',
+    konnector: 'caissedepargne1'
+  })
+  const event = await waitForRealtimeEvent(
+    client,
+    jobResponse.data.attributes,
+    'result',
+    30 * 1000
+  )
+  return event.data.result
+}
+
+const savePaymentCreation = async (client, paymentCreation) => {
+  const label = get(paymentCreation, 'instructions[0].label')
+  const beneficiaryAccount = get(
+    paymentCreation,
+    'instructions[0].beneficiary.identification'
+  )
+  const beneficiaryLabel = get(
+    paymentCreation,
+    'instructions[0].beneficiary.label'
+  )
+  const executionDate = get(paymentCreation, 'register_date')
+  const executionDateType = get(
+    paymentCreation,
+    'instructions[0].execution_date_type'
+  )
+  const currency = get(paymentCreation, 'instructions[0].currency.id')
+  try {
+    await client.save({
+      _type: TRANSACTION_DOCTYPE,
+      date: executionDate,
+      dateType: executionDateType,
+      transferId: paymentCreation.id,
+      status: paymentCreation.status,
+      label,
+      beneficiaryLabel,
+      beneficiaryAccount,
+      currency
+    })
+  } catch (e) {
+    log('error', "Can't save transaction with payment object")
+    log('error', e)
+  }
+}
+
+export const createPaymentCreation = async ({ client, payment, biPayment }) => {
+  const { url, token, clientRedirectUri } = biPayment
+  const payload = makePayload(payment, clientRedirectUri)
```

Comment:
OK got it for the previous question. But other question. Where does this `clientRedirectUri` is defined?

Reply:
There is in our global secret database, accessible via the banking connector.